### ISSUE TITLE
armv7h mirrorlist?

### ISSUE THREAD

**Issue by bammerlaan** [arch4edu/arch4edu#246]
I'm almost certain I'm doing something wrong, but hopefully you can point me in the right direction. I'm trying to add arch4edu as an extra repo to my Raspberry Pi 4 archlinux server (in order to use your Jellyfin packages). 

I added the arch4edu repo to my pacman.conf like so:

```
#
# /etc/pacman.conf
#
# See the pacman.conf(5) manpage for option and repository directives

#
# GENERAL OPTIONS
#
[options]
# The following paths are commented out with their default values listed.
# If you wish to use different paths, uncomment and update the paths.
#RootDir     = /
#DBPath      = /var/lib/pacman/
#CacheDir    = /var/cache/pacman/pkg/
#LogFile     = /var/log/pacman.log
#GPGDir      = /etc/pacman.d/gnupg/
#HookDir     = /etc/pacman.d/hooks/
HoldPkg     = pacman glibc
#SyncFirst = archlinux-keyring
#XferCommand = /usr/bin/curl -L -C - -f -o %o %u
#XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
#CleanMethod = KeepInstalled
Architecture = armv7h

# Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
#IgnorePkg   =
#IgnoreGroup =

#NoUpgrade   =
#NoExtract   =

# Misc options
#UseSyslog
Color
#NoProgressBar
CheckSpace
#VerbosePkgLists
#ParallelDownloads = 5

# By default, pacman accepts packages signed by keys that its local keyring
# trusts (see pacman-key and its man page), as well as unsigned packages.
SigLevel    = Required DatabaseOptional
LocalFileSigLevel = Optional
#RemoteFileSigLevel = Required

# NOTE: You must run `pacman-key --init` before first using pacman; the local
# keyring can then be populated with the keys of all official Arch Linux ARM
# packagers with `pacman-key --populate archlinuxarm`.

#
# REPOSITORIES
#   - can be defined here or included from another file
#   - pacman will search repositories in the order defined here
#   - local/custom mirrors can be added here or in separate files
#   - repositories listed first will take precedence when packages
#     have identical names, regardless of version number
#   - URLs will have $repo replaced by the name of the current repo
#   - URLs will have $arch replaced by the name of the architecture
#
# Repository entries are of the format:
#       [repo-name]
#       Server = ServerName
#       Include = IncludePath
#
# The header [repo-name] is crucial - it must be present and
# uncommented to enable the repo.
#

# The testing repositories are disabled by default. To enable, uncomment the
# repo name header and Include lines. You can add preferred servers immediately
# after the header, and they will be used before the default mirrors.

[core]
Include = /etc/pacman.d/mirrorlist

[extra]
Include = /etc/pacman.d/mirrorlist

[alarm]
Include = /etc/pacman.d/mirrorlist

[aur]
Include = /etc/pacman.d/mirrorlist

#[arch4edu]
#Server = https://pkg.fef.moe/arch4edu/$arch

# An example of a custom package repository.  See the pacman manpage for
# tips on creating your own repositories.
#[custom]
#SigLevel = Optional TrustAll
#Server = file:///home/custompkgs

```

Running `pacman -Syy` results in the following, though:

```
> sudo pacman -Syy
:: Synchronizing package databases...
 core                                         237.6 KiB   200 KiB/s 00:01 [#########################################] 100%
 extra                                          9.0 MiB   891 KiB/s 00:10 [#########################################] 100%
 alarm                                         95.6 KiB   233 KiB/s 00:00 [#########################################] 100%
 aur                                            8.5 KiB  3.65 KiB/s 00:02 [#########################################] 100%
 arch4edu.db failed to download
error: failed retrieving file 'arch4edu.db' from pkg.fef.moe : The requested URL returned error: 404
error: failed to synchronize all databases (failed to retrieve some files)
```

Where am I going wrong? I suspect it's something to do with not supporting my architecture, but then again, arch4edu supports ARM devices, right?

**Comment by petronny** [comment]
Hi, we've dropped the armv7h support when we changed our build system. The kernel used in our current aarch64 builder isn't complied with 32-bit (ia32) support.

To add armv7h back, we need:
1. a native armv7h builder or an aarch64 builder with a kernel that has ia32 support.
2. a few weeks to make some changes in the build system to build and push armv7h packages again.
3. some confidence that archlinuxarm will not drop the armv7h support soon.

Personally, I believe alarm will drop the armv7h support maybe in a year or two. So I'm afraid it may not worth the effort to add it back.
And just curious, is there any reason to keep you using the armv7h alarm than the aarch64 one on rpi4?

**Comment by bammerlaan** [comment]
Ahh, I see. Alright, that makes sense.

> And just curious, is there any reason to keep you using the armv7h alarm than the aarch64 one on rpi4?

Not particularly. In fact, I admit I wasn't aware that was a possibility. Back when I installed alarm on the device I just went with the installation instructions on their site. I probably read over the fact that I _could_ install AArch64. I regret that now... I'll have to reinstall alarm on it soon now, I suppose. Thank you for your reply!
